src/core/dep-graph.ts

```diff
@@ -327,17 +306,26 @@ class DepGraphImpl implements types.DepGraphInternal {
     return node;
   }
 
-  private pathsFromNodeToRoot(nodeId: string): types.PkgInfo[][] {
+  private pathsFromNodeToRoot(
+    nodeId: string,
+    ancestors: string[] = [],
```

Comment:
Is there a reason you chose an Array instead of a Set?

Reply:
Consistency as we talked